src/reader.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#ifndef INCLUDE_reader_h__
+#define INCLUDE_reader_h__
+
+#include "common.h"
+
+typedef struct git_reader git_reader;
+
+struct git_reader {
+	int (*read)(git_buf *out, git_reader *reader, const char *filename);
+	void (*free)(git_reader *reader);
+};
+
+extern int git_reader_for_nothing(
```

Comment:
Curious: any reason you're using `extern` directly ? I thought `GIT_EXTERN` was mandatory because of MSVC compatibility…

Reply:
`GIT_EXTERN` is what we use for things in the public api.  It will do all the decoration for DLL exports, as you mentioned.  For private functions within the library, we should simply use `extern`.